test/controllers/users_controller_test.rb

```diff
@@ -0,0 +1,37 @@
+require 'test_helper'
+
+class UsersControllerTest < ActionDispatch::IntegrationTest
+  test 'should get /users endpoint' do
+    get '/users'
+    assert_response :success
+  end
+
+  test '/users should retrieve all users as json' do
+    users = ['John', 'Polly']
+    users.each { |user| User.create(name: user)}
+
+    get '/users'
+
+    users_response = JSON.parse(@response.body, symbolize_names: true)
+    response_names = users_response.map {|user| user[:name]}
+
+    assert_equal response_names, users
+    assert_equal 2, users_response.count
+  end
+
+  test 'should get /users/1 endpoint' do
+    User.create(name: 'Test')
+    all_users = User.all
+    get "/users/#{all_users[0]['id']}"
+    assert_response :success
+  end
+
+  test '/users/1 should retrieve a single user' do
+    user_name = 'Maggie'
+    User.create(name: user_name)
```

Comment:
Instead of this, you can do: 

```suggestion
    user = User.create(name: user_name)
    get "/users/#{user['id']}"
```